docs/documentation/generation/Overview.md

```diff
@@ -0,0 +1,110 @@
+## Summary
+MOOSE includes tools for building autogenerating web-sites for that documents your application. Included
+in this tool is the ability to:
+
+* Add detailed descriptions of custom systems and objects using a [Moose Style Markdown](MooseFlavoredMarkdown.md),
+which has useful features designed for documenting source code and input file syntax;
+* Generate pages bassed on class descriptions and input parameter documentation included in your source code;
+* Generate links to examples and source code that use or inherit from your objects;
+* Create summary tables of objects included in your application; and
+* Include documentation from the MOOSE framework, modules, or other applications that are registered in your application.
+
+## Documenting Applications
+### (1) Source Code Documentation
+The first step is to add documentation for your application in the `validParams` function.
+
+#### Class Description
+A short description should be supplied in each `validParams` function in your code. For example, in
+the [Diffusion](../systems/framework/Kernels/Diffusion.md) object the following class
+description, which is added to the generated documentation as the first sentance of the page.
+
+![](framework/src/kernels/Diffusion.C line=addClassDescription)
+
+#### Parameter Descriptions
+A description of each parameter should also be provided when they are added with the various add methods
+of the `InputParameters` object. For example, in the [FunctionIC](../systems/framework/ICs/FunctionIC.md)
+the following parameter documentation is added, which is then present in the parameter summary table of the
+generated site.
+
+![](framework/src/ics/FunctionIC.C line=addRequiredParam)
+
+### (2) Configuration
+The documentation generation system for MOOSE is built upon [mkdocs](www.mkdocs.org), thus it is configured
+using [YAML](http://yaml.org) files. For MOOSE there are two configuration files to setup: a source and application
+level configuration.
+
+#### Source Configure File
+Inside each source code directory that MOOSE code exists a "config.yml" file must exist. In general, most
```

Comment:
Change sentence to "A config.yml file must exist in each directory containing MOOSE source."